src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/AttributeDictionary.cs

```diff
@@ -0,0 +1,653 @@
+                }
+                else
+                {
+                    Set(index, item);
+                }
+            }
+        }
+
+        public int Count { get; private set; }
+
+        public bool IsReadOnly
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public ICollection<string> Keys
+        {
+            get
+            {
+                return new KeyCollection(this);
+            }
+        }
+
+        public ICollection<string> Values
+        {
+            get
+            {
+                return new ValueCollection(this);
+            }
+        }
+
+        IEnumerable<string> IReadOnlyDictionary<string, string>.Keys
+        {
+            get
+            {
+                return new KeyCollection(this);
+            }
+        }
+
+        IEnumerable<string> IReadOnlyDictionary<string, string>.Values
+        {
+            get
+            {
+                return new ValueCollection(this);
+            }
+        }
+
+        private KeyValuePair<string, string> Get(int index)
+        {
+            if (index >= Count)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            switch (index)
+            {
+                case 0:
+                    return _items0;
+                case 1:
+                    return _items1;
+                case 2:
+                    return _items2;
+                case 3:
+                    return _items3;
+                default:
+                    return _items[index - FixedEntryCount];
+            }
+        }
+
+        private void Set(int index, KeyValuePair<string, string> value)
+        {
+            if (index > Count)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            if (index == Count)
+            {
+                Count++;
+            }
+
+            switch (index)
+            {
+                case 0:
+                    _items0 = value;
+                    return;
+                case 1:
+                    _items1 = value;
+                    return;
+                case 2:
+                    _items2 = value;
+                    return;
+                case 3:
+                    _items3 = value;
+                    return;
+                default:
+                    if (_items == null)
+                    {
+                        _items = new List<KeyValuePair<string, string>>();
+                    }
+
+                    _items[index - FixedEntryCount] = value;
+                    return;
+            }
+        }
+
+        private void Insert(int index, KeyValuePair<string, string> value)
+        {
+            if (index > Count)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            if (_items == null)
+            {
+                _items = new List<KeyValuePair<string, string>>();
+            }
```

Comment:
Not really relevant anymore because I'm going with the simple list version for now.